app.rb

```diff
@@ -0,0 +1,195 @@
+    @next_id += 1
+    rtn_id
+  end
+
+  def run
+    puts 'Welcome to School Library App!'
+    options_case
+  end
+
+  def list_all_books
+    @books.each do |book|
+      puts "  #{book.title} by #{book.author}"
+      puts '  rented by:'
+      book.rentals.each do |rent|
+        puts "    #{rent.person.name} at #{rent.date}"
+      end
+    end
+  end
+
+  def add_book
+    print 'title: '
+    title = gets.chomp
+    print 'author: '
+    author = gets.chomp
+    @books.push(Book.new(title, author))
+  end
+
+  def list_all_persons
+    puts ' ---- Students'
+    @students.each do |student|
+      put_str = "  [#{student.id}] #{student.name} has #{student.age}"
+      put_str += " years old #{student.can_use_services? ? 'and can use services' : ''}"
+      puts put_str
+    end
+    puts ' ---- Teachers'
+    @teachers.each do |teacher|
+      put_str = "  [#{teacher.id}] #{teacher.name} has #{teacher.age}"
+      put_str += " years old specialized in: #{teacher.specialization}"
+      puts put_str
+    end
+  end
+
+  def person_info_input
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    { age: age.to_i, name: name }
+  end
+
+  def print_all_rentals
+    @rentals.each do |rent|
+      puts "#{rent.book.title} rented by #{rent.person.name} at #{rent.date}"
+    end
+  end
+
+  def add_rental
+    puts 'Select a Book from the following list by number'
+    @books.each_with_index do |book, index|
+      puts "[#{index}] #{book.title}"
+    end
+    book = @books[gets.chomp.downcase.to_i]
+
+    persons = @teachers + @students
+
+    puts 'Select a person from the following list by number'
+    persons.each_with_index do |person, index|
+      puts "[#{index}] #{person.name}"
+    end
+    person = persons[gets.chomp.downcase.to_i]
+
+    print 'Date: '
+    date = gets.chomp.downcase
+
+    @rentals.push(Rental.new(date, person, book))
+  end
+
+  def add_person
+    print 'Do you want to create a student [1] or teacher [2] Input the Number: '
+    number = gets.chomp.downcase
+
+    case number
+    when '1' # STUDENT
+      person_info = person_info_input
+
+      print 'Has parrent permission? [Y/N]: '
+      permission = gets.chomp.downcase == 'y'
+
+      @students.push(Student.new(
+                       generate_next_id,
+                       person_info[:age],
+                       person_info[:name],
+                       parent_permission: permission
+                     ))
+    when '2' # TEACHER
+      person_info = person_info_input
+
+      print 'Specialization: '
+      specialization = gets.chomp
+
+      @teachers.push(Teacher.new(
+                       generate_next_id,
+                       person_info[:age],
+                       specialization,
+                       person_info[:name]
+                     ))
+
+    else
+      "You gave me #{option} -- I have no idea what to do with that.\n"
+    end
+
+    # @books.push(Book.new(title, author))
```

Comment:
- [ ] Please remove all the commented code from your codebase in order to make your code clean. :smile: